src/main/java/com/intuit/graphql/orchestrator/schema/transform/RenameTransformer.java

```diff
@@ -0,0 +1,165 @@
+    Pair<String, String> renamedPair = getRenamedPair(typeDefinition, renameDirective);
+    addRenamedPairToMap(renamedPair, renamedTypeMap);
+    typeDefinition.setName(renamedPair.getKey());
+  }
+
+  private void addFieldToMap(String parentName, FieldDefinition fieldDefinition,
+                             Directive renameDirective, Map<String, String> renamedFieldMap) {
+    Pair<String, String> originalRenamedPair = getRenamedPair(fieldDefinition, renameDirective);
+    String originalAlias = originalRenamedPair.getKey();
+
+    String metadataKey = getRenameKey(parentName, originalRenamedPair.getKey(), (OperationType.get(parentName.toLowerCase()) != null));
+    Pair<String, String> renamedPair = Pair.of(metadataKey, originalRenamedPair.getValue());
+
+    addRenamedPairToMap(renamedPair, renamedFieldMap);
+    fieldDefinition.setName(originalAlias);
+  }
+
+  private Pair<String, String> getRenamedPair(FieldDefinition fieldDefinition, Directive renameDirective){
+    String originalName = fieldDefinition.getName();
+    String alias = renameDirective.getArguments().stream()
+            .filter(argument -> argument.getName().equals(RENAME_ARGUMENT_NAME))
+            .findFirst()
+            .map(toArgument -> toArgument.getValueWithVariable().getStringValue())
+            .orElse(null);
+
+    validateToValue(alias);
+
+    return Pair.of(alias, originalName);
+  }
+
+  private Pair<String, String> getRenamedPair(TypeDefinition typeDefinition, Directive renameDirective){
+    String originalName = typeDefinition.getName();
+    String alias = renameDirective.getArguments().stream()
+            .filter(argument -> argument.getName().equals(RENAME_ARGUMENT_NAME))
+            .findFirst()
+            .map(toArgument -> toArgument.getValueWithVariable().getStringValue())
+            .orElse(null);
+
+    validateToValue(alias);
+
+    return Pair.of(alias, originalName);
+  }
+
+  private Pair<String, String> getRenamedPair(TypeExtensionDefinition typeDefinition, Directive renameDirective){
+    String originalName = typeDefinition.getName();
+    String alias = renameDirective.getArguments().stream()
+            .filter(argument -> argument.getName().equals(RENAME_ARGUMENT_NAME))
+            .findFirst()
+            .map(toArgument -> toArgument.getValueWithVariable().getStringValue())
+            .orElse(null);
+
+    validateToValue(alias);
+
+    return Pair.of(alias, originalName);
+  }
+
+  private void validateToValue(String toValue) {
+    if(StringUtils.isBlank(toValue)) {
+      throw new RuntimeException("to argument is empty for %s");
+    }
+
+    if(StringUtils.containsWhitespace(toValue)) {
+      throw new RuntimeException("to argument for cannot %s contain whitespace");
+    }
+  }
+
+  private void addRenamedPairToMap(Pair<String, String> renamedPair, Map<String, String> renameMap) {
+    if(renameMap.containsKey(renamedPair.getValue())) {
+      throw new RuntimeException("Multiple renames with the same name");
```

Comment:
Please check that it boils down to an appropriate exception for the register endpoint in data api